### PR TITLE
feat: mvn kalix:run with proxy in dev mode

### PR DESCRIPTION
**Pull request by octonato**
This is working now. 
Still needs some clean-ups and better thread management.

### CODE REVIEW

**Comment by octonato** [comment]
This is ready for review. 

We still need to adapt the README files, the poms and the documentation, but I will wait for a round of feedback before moving forward with docs. 

Note, this only covers maven (protobuf and code-first). For Scala we will need another PR. 
Also, we should not merge this before we have proxy 1.1.7. Running `mvn kalix:run` with pre-1.1.7 images can be very very slow.

File: devtools/src/main/scala/kalix/devtools/impl/KalixProxyContainer.scala [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+  val logger = LoggerFactory.getLogger(classOf[KalixProxyContainer])
+
+  def apply(config: KalixProxyContainerConfig): KalixProxyContainer = {
+
+    val dockerImage: DockerImageName =
+      if (config.proxyImage.trim.nonEmpty)
+        DockerImageName.parse(config.proxyImage)
+      else
+        DockerImageName.parse(BuildInfo.proxyImage).withTag(BuildInfo.proxyVersion)
```
Comment (octonato) [inline devtools/src/main/scala/kalix/devtools/impl/KalixProxyContainer.scala:47]:
```suggestion
    val dockerImage: DockerImageName = DockerImageName.parse(config.proxyImage)
```

**Comment by octonato** [comment]
> Minor suggestion, I wonder if we can improve the logging somehow?
> 
> ```
> [INFO] Creating container for image: gcr.io/kalix-public/kalix-proxy:1.1.7-M1
> 14:45:36.532 INFO  k.s.b.KalixReactiveWebServerFactory - Starting Kalix ReactiveWebServer...
> 14:45:36.532 INFO  k.spring.impl.KalixSpringApplication - Starting Kalix Application...
> 14:45:36.869 INFO  com.example.Main - Started Main in 1.601 seconds (process running for 1.858)
> [INFO] Container gcr.io/kalix-public/kalix-proxy:1.1.7-M1 is starting: bebd42a08da8dc6eb8565c7c0412dd28d8cf9a7d8e32c23879e343414e736f8e
> [INFO] Container gcr.io/kalix-public/kalix-proxy:1.1.7-M1 started in PT0.898192S
> [INFO] STDOUT: 12:45:38.163 INFO  kalix.proxy.KalixProxyMain - Starting Kalix Proxy version [1.1.7-M1] Dev Mode
> [INFO] STDOUT: 12:45:38.179 INFO  kalix.proxy.telemetry.Telemetry - JVM [OpenJDK Runtime Environment 17.0.3+7], max heap [1964 MB], processors [5]
> [INFO] STDOUT: 12:45:38.494 INFO  kalix.proxy.DiscoveryManager - Connecting to [host.testcontainers.internal:8080] to discover the user function specification
> ```
> 
> proxy logs go to with `[INFO] STDOUT...` whereas uf logs are printed without any prefix.

Yeah, I also don't like it. 
What is happening here is that the logs are captured by testcontainers and then forwarded locally. I need to search further to see if there is anything else we can do.

**Review by efgpinto — APPROVED** [review]
LGTM, great work!

some questions, for my own understanding:
- we still require `mvn compile kalix:run`, right?
- auto-reload after code modifications is not included yet, correct?

**Comment by octonato** [comment]
> LGTM, great work!
> 
> some questions, for my own understanding:
> 
> * we still require `mvn compile kalix:run`, right?

No, I pushed a new commit this afternoon that triggers compilation when running `kalix:run`. 

Calling `mvn clean kalix:run` has the same effect of `mvn clean compile kalix:run`.

> * auto-reload after code modifications is not included yet, correct?

Correct. Auto-reload will be a separated task. We need to investigate it further. I was experimenting with the Spring dev-tools. That will probably work easily now and the proxy will just keep working in the background. 

However, that won't work for Java/Scala Protobuf. So, maybe we should have our own auto-reload tool that we can then user everywhere.

**Review by aludwiko — APPROVED** [review]
works quite well locally :)

**Comment by octonato** [comment]
Yep, proxy is on it's way. But since we decide to keep this feature under the radar for now, we can already merge. Apple M1 users should tests by manually passing 1.1.7 image.

**Pull request merged**